nmtwizard/preprocess/operators/noise.py

```diff
@@ -41,17 +45,49 @@ def __init__(self, config, *args):
             config = source_config
         self._drop_word_prob = config.get("drop_word_prob", 0)
         self._drop_space_prob = config.get("drop_space_prob", 0)
+        self._insert_space_prob = config.get("insert_space_prob", 0)
         self._drop_char_prob = config.get("drop_char_prob", 0)
         self._duplicate_char_prob = config.get("duplicate_char_prob", 0)
         self._swap_char_prob = config.get("swap_char_prob", 0)
+        self._substitute_char_prob = config.get("substitute_char_prob", 0)
+        self._add_marker = config.get("add_marker", 0)
 
     def _preprocess_tu(self, tu, *args):
+        tu = self._apply_space_insertion_noise(tu)
         src_tok = tu.src_tok
         tokens = src_tok.token_objects
+        original_tokens = copy.deepcopy(tokens)
         new_tokens = [self._apply_word_noise(tokens[0])]
         tu.src_tok = (src_tok.tokenizer, new_tokens)
+        if self._add_marker and new_tokens != original_tokens:
+            tu.replace_tokens_side("source", (0, 0, ["｟mrk_noisy｠"]))
```

Comment:
Looks like this check doesn't account for space insertion. Is it expected ?

Reply:
No, it was not expected, thanks for seeing that!